include/deal.II/lac/vector_memory.templates.h

```diff
@@ -42,18 +43,12 @@ inline
 GrowingVectorMemory<VectorType>::Pool::~Pool()
 {
   // Nothing to do if memory was unused.
-  if (data == nullptr) return;
-
-  // First, delete all remaining
-  // vectors. Actually, there should
-  // be none, if there is no memory
-  // leak
-  for (typename std::vector<entry_type>::iterator i=data->begin();
-       i != data->end();
-       ++i)
-    {
-      delete i->second;
-    }
+  if (data == nullptr)
+    return;
+
+  // delete the 'data' object. this also releases all vectors
+  // that are pointed to by the std::unique_ptrs
+  data->clear();
   delete data;
```

Comment:
I wondered about that as well. I don't recall, so I left that part alone. I have a suspicion that it had to do with explicitly managing when things are destroyed. For example, the destructor of that class does not delete the pointer. This creates a memory leak (but then the owner is a `static` object anyway, whose destructor is only run at the end of the program) that I can only imagine is on purpose. But I can not say that I would recall (or even remember who wrote this).